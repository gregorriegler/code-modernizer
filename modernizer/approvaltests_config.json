{
  "reporter": "PythonNativeReporter",
  "subdirectory": "approved_files"
}
