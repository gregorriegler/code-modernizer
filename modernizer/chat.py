from dataclasses import dataclass, field
from typing import List, Dict

@dataclass(frozen=True)
class Chat:
    _messages: List[Dict[str, str]] = field(default_factory=list)

    def userSays(self, content: str) -> 'Chat':
        return self.add("user", content)

    def add(self, role: str, content: str) -> 'Chat':
        new_messages = list(self._messages)
        new_messages.append({"role": role, "content": content})
        return Chat(new_messages)
    
    def to_list(self) -> List[Dict[str, str]]:
        return list(self._messages)
    
    def __len__(self) -> int:
        return len(self._messages)
    
    def __iter__(self):
        return iter(self._messages)
    
    def __str__(self) -> str:
        return str(self._messages)