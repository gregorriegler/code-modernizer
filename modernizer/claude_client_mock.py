def message_claude(messages, system_prompt):
    print("\n[MOCK] Claude would have sent this to the API:")
    print(f"System Prompt: {system_prompt[:100]}..." if system_prompt else "No system prompt")
    print(f"Last message: {messages[-1]['content'][:200]}..." if messages else "No messages")
    return "This is a mock response from Claude. No actual API call was made."
