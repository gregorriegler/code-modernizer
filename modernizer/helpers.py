import sys

def read_file(filename):
    """Read content from a file, handling errors gracefully."""
    try:
        with open(filename, 'r') as f:
            return f.read().strip()
    except FileNotFoundError:
        print(f"Error: {filename} not found", file=sys.stderr)
        sys.exit(1)
    except Exception as e:
        print(f"Error reading {filename}: {e}", file=sys.stderr)
        sys.exit(1)