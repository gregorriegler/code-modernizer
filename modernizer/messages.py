from dataclasses import dataclass, field
from typing import List, Dict, Any

@dataclass(frozen=True)
class Messages:
    _messages: List[Dict[str, str]] = field(default_factory=list)

    def add(self, role: str, content: str) -> 'Messages':
        new_messages = list(self._messages)
        new_messages.append({"role": role, "content": content})
        return Messages(new_messages)
    
    def to_list(self) -> List[Dict[str, str]]:
        return list(self._messages)
    
    def __len__(self) -> int:
        return len(self._messages)
    
    def __iter__(self):
        return iter(self._messages)
    
    def __str__(self) -> str:
        return str(self._messages)
