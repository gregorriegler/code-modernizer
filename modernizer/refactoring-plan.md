# Refactoring Plan

## Identified Improvements

### mutation_tool.py
- [x] Long method: `_format_mutation_output` is doing too many things (parsing JSON, counting mutants, formatting output)
- [x] Extract method for mutant counting and statistics calculation
- [x] Extract method for survived mutant formatting
- [x] Magic numbers: hardcoded mutation score calculation could be extracted to a constant or method
- [x] Improve variable naming: `stryker_data` could be more descriptive like `mutation_report`

### mutation_tool_test.py
- [ ] Long test method: `test_parse_stryker_output` is testing multiple aspects (parsing, counting, formatting)
- [ ] Extract helper method for creating mock Stryker JSON data
- [ ] Consider splitting into separate tests for different aspects (summary calculation, survived mutants formatting)