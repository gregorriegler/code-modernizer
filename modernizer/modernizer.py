#!/usr/bin/env python

import argparse
import json
import os
import sys

from claude_client import message_claude
from helpers import *
from chat import Chat
from tools import ToolLibrary


def load_session(session_file):
    """Load conversation history from session file."""
    if not os.path.exists(session_file):
        return Chat()
    
    try:
        with open(session_file, 'r') as f:
            chat_data = json.load(f)
            return Chat(chat_data if isinstance(chat_data, list) else [])
    except (json.JSONDecodeError, Exception) as e:
        print(f"Warning: Could not load session file {session_file}: {e}", file=sys.stderr)
        return Chat()

def save_session(chat):
    session_file = "claude-session.json"
    try:
        with open(session_file, 'w') as f:
            json.dump(chat.to_list(), f, indent=2)
    except Exception as e:
        print(f"Warning: Could not save session file {session_file}: {e}", file=sys.stderr)

def get_system_prompt():
    script_dir = os.path.dirname(os.path.abspath(__file__))
    system_prompt_path = os.path.join(script_dir, "system-prompt.md")
    system_prompt = None
    if os.path.exists(system_prompt_path):
        system_prompt = read_file(system_prompt_path)
    return system_prompt


def main():
    parser = argparse.ArgumentParser(description="Claude API CLI with session support")
    parser.add_argument("--new", action="store_true", help="Start a new session (clear history)")
    parser.add_argument("message", nargs="*", help="Message to send to Claude")
    
    args = parser.parse_args()
    start_message = None
    if len(sys.argv) > 1:
        start_message = " ".join(sys.argv[1:])
    if args.message:
        start_message = " ".join(args.message)
    new = args.new

    start_chat(start_message, new, message_claude, save_session=save_session)


def start_chat(start_message, new, message_claude, rounds=999999, save_session=save_session):
    system_prompt = get_system_prompt()
    tools = ToolLibrary()

    chat = Chat() if new else load_session("claude-session.json")
    print("Starting new session" if new else "Continuing session")
    
    if start_message:
        chat = chat.add("user", start_message)

    for _ in range(rounds):
        answer = message_claude(chat.to_list(), system_prompt)
        print(f"\nClaude: {answer}")
        chat = chat.add("assistant", answer)

        try:
            user_input = input("\nPress Enter to continue or type a message to add: ")
            if user_input.strip():
                chat = chat.add("user", user_input)
                continue
        except EOFError:
            print("\nExiting...")
            break
        except KeyboardInterrupt:
            print("\nExiting...")
            break

        _, tool_result = tools.parse_and_execute(answer)

        if tool_result:
            chat = chat.add("user", tool_result)

        save_session(chat)


if __name__ == "__main__":
    main()