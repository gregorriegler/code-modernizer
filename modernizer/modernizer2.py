import os

from claude_client import message_claude
from helpers import *
from chat import Chat, load_chat, save_chat
from tools import ToolLibrary


def get_system_prompt():
    script_dir = os.path.dirname(os.path.abspath(__file__))
    system_prompt_path = os.path.join(script_dir, "system-prompt.md")
    system_prompt = None
    if os.path.exists(system_prompt_path):
        system_prompt = read_file(system_prompt_path)
    return system_prompt


def start_chat(start_message, new, message_claude, rounds=999999, save_chat=save_chat):
    system_prompt = get_system_prompt()
    tools = ToolLibrary()

    chat = Chat() if new else load_chat("claude-session.json")
    print("Starting new session" if new else "Continuing session")
    
    if start_message:
        chat = chat.userSays(start_message)

    for _ in range(rounds):
        answer = message_claude(chat.to_list(), system_prompt)
        print(f"\nClaude: {answer}")
        chat = chat.assistantSays(answer)

        try:
            user_input = input("\nPress Enter to continue or type a message to add: ")
            if user_input.strip():
                chat = chat.userSays(user_input)
                continue
        except EOFError:
            print("\nExiting...")
            break
        except KeyboardInterrupt:
            print("\nExiting...")
            break

        _, tool_result = tools.parse_and_execute(answer)

        if tool_result:
            chat = chat.userSays(tool_result)

        save_chat(chat)