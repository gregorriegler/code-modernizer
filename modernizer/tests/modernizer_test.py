import builtins
import os
import re
import sys
import pytest
from approvaltests import verify, verify_as_json
from approvaltests.reporters.report_with_beyond_compare import ReportWithWinMerge
from approvaltests import set_default_reporter
from approvaltests.scrubbers import create_regex_scrubber
from approvaltests import Options
sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
from modernizer.modernizer import start_chat

@pytest.fixture(scope="session", autouse=True)
def set_default_reporter_for_all_tests() -> None:
    set_default_reporter(ReportWithWinMerge())

def create_temp_file(tmp_path, filename, contents):
    temp_file = tmp_path / filename
    temp_file.write_text(contents)
    return temp_file

def enter(_):
    return "\n"

def keyboard_interrupt(_):
    raise KeyboardInterrupt()
    
def run_chat_test(capsys, input_stub, message, answer):
    builtins.input = input_stub
    claude_stub = lambda messages, system_prompt: answer

    saved_messages = "None"
    def save_session(messages):
        nonlocal saved_messages
        saved_messages = "\n".join(str(msg['role'] + ": " + msg['content']) for msg in messages)
    
    try:
        start_chat(message, new=True, message_claude=claude_stub, rounds=1, save_session=save_session)
    except KeyboardInterrupt:
        pass
        
    captured = capsys.readouterr()
    return f"""# Standard out:
{captured.out}

# Saved messages:
{saved_messages}"""

def test_start_chat_with_new_session(capsys):
    result = run_chat_test(capsys, input_stub=enter, message="Test message", answer="Test answer")
    verify(result)

def test_abort(capsys):
    result = run_chat_test(capsys, input_stub=keyboard_interrupt, message="Test message", answer="Test answer")
    verify(result)

def test_tool_cat(capsys, tmp_path):
    temp_file = create_temp_file(tmp_path, "testfile.txt", "Hello world")
    
    result = run_chat_test(capsys, input_stub=enter, message="Test message", answer=f"/cat {temp_file}")
    
    temp_path_scrubber = create_regex_scrubber('/cat.*', '/cat /tmp/path/testfile.txt')
    verify(result, options=Options().with_scrubber(temp_path_scrubber))