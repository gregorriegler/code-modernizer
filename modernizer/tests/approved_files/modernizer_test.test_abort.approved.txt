Starting new session

Claude: Test answer

Exiting...
